compiler/testData/codegen/bytecodeText/redundantValInitializer.kt

```diff
@@ -0,0 +1,30 @@
+package a
+
+class A {
+    companion object {
+        const val constInt: Int = 0
```

Comment:
I'm not sure but maybe we need also tests non const vals?

Reply:
I figured that I covered both by having the "my*" fields pointing to the constants in the companion object. Anyway to be explicit, I added fields directly initialized to literal constants.

I also added tests for fields.